.github/workflows/e2e_tests.yaml

```diff
@@ -0,0 +1,190 @@
+name: e2e
+
+on:
+  pull_request:
+    branches:
+      - main
+      - 'release-*'
+    paths-ignore:
+      - 'docs/**'
+      - '**.adoc'
+      - '**.md'
+      - 'LICENSE'
+  push:
+    branches:
+      - main
+      - 'release-*'
+    paths-ignore:
+      - 'docs/**'
+      - '**.adoc'
+      - '**.md'
+      - 'LICENSE'
+
+concurrency:
+  group: ${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  kubernetes:
+
+    runs-on: ubuntu-20.04
```

Comment:
Is there any reason to use specific version and not `ubuntu-latest`?

Reply:
I've experienced workflows failing over night when using rolling tags, and found it sometimes difficult to realise the root cause was the implicit upgrade of a dependency. I think build / test run reproducibility largely overweights the cost of explicit upgrades, at least in the case of an OS upgrade.